src/adera/Machines/Rocket.cpp

```diff
@@ -125,7 +124,8 @@ void SysMachineRocket::update_physics(ActiveScene& rScene)
         bool fail = false;
         for (auto& resource : machine.m_resourceLines)
         {
-            auto const* src = m_scene.reg_try_get<MachineContainer>(resource.m_sourceEnt);
+            auto const* pipe = resource.m_source.get_if<wiretype::Pipe>();
+            auto const* src = rScene.reg_try_get<MachineContainer>(pipe->m_source);
```

Comment:
`pipe` here can still be null, and causes a crash. Also use an `== nullptr` check instead of an implicit boolean conversion.

Another issue, on line 135, the ! operator is being applied to m_quantity, not the entire if statement. Put brackets, or even better, change to `0 ==`. Silly enough `(!m_quantity) > 0` still works.

Reply:
you probably want something like

if(auto const* pipe = resource.m_source.get_if<wiretype::Pipe>(); nullptr != pipe)
{
    auto const* src = rScene.reg_try_get<MachineContainer>(pipe->m_source);

    .... other stuff
}

Reply:
And yea, line 135 is wrong. Operator precedence is a bitch.

Had a guy ask me why i'm so insistent on parenthesis usages the other day, and this is basically the reason.